### PR TITLE
fix(lazy-loading): fixed webpack filename clashing issue

### PR DESCRIPTION
**Pull request by bishnubista**
#### _Describe your changes in detail_
Library name is changed in `webpack.client.js` to fix issue of web-pack filename clashing during lazy loading.

#### _Why is this change required? What issue does it resolve?_
To fix lazy loading issue caused by web-pack filename clash.

#### Check boxes that apply:
- [ X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (adding or updating documentation)
- [ ] Dependency update
- [ ] Security update

#### Check boxes that apply:
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation and I have updated the documentation accordingly.
- [ ] These changes should be applied to a maintenance branch.
- [ ] I have added the Apache 2.0 license header to any new files created.

### CODE REVIEW

File: packages/one-app-bundler/webpack/module/webpack.client.js [hunk 1]
```diff
@@ -36,6 +36,7 @@ const packageRoot = process.cwd();
 const { packageJson } = readPkgUp.sync();
 const { version, name } = packageJson;
 
+const holocronModuleName = `_${name.replace(/-/g, '_')}`;
```
Comment (Matthew-Mallimo) [inline packages/one-app-bundler/webpack/module/webpack.client.js:39]: Why are we replacing the dashes?

**Comment by 10xLaCroixDrinker** [comment]
Shouldn't this also include an updated version of the plugin?

**Comment by bishnubista** [comment]
> Shouldn't this also include an updated version of the plugin?

Yes. I will change this to draft till `holocron` changes gets merged. Once that gets published I will update version and put it to review.

File: packages/one-app-bundler/webpack/module/webpack.client.js (hunk 1 above)
Comment (bishnubista) [reply packages/one-app-bundler/webpack/module/webpack.client.js]: If used `hyphens` Terser plugin throws following error. 

`Unexpected token: operator (-)`

**Comment by bishnubista** [comment]
> Shouldn't this also include an updated version of the plugin?

Yes. It is updated to 1.2.2.

File: packages/one-app-bundler/__tests__/webpack/module/webpack.client.spec.js [hunk 2]
```diff
@@ -13,12 +13,13 @@
...
 
 /* eslint-disable global-require */
+const { sync } = require('read-pkg-up');
 const { validateWebpackConfig } = require('../../../test-utils');
 const getConfigOptions = require('../../../utils/getConfigOptions');
 const configGenerator = require('../../../webpack/module/webpack.client');
 
 jest.mock('../../../utils/getConfigOptions', () => jest.fn(() => ({ purgecss: {} })));
-
+jest.spyOn(process, 'cwd').mockImplementation(() => __dirname.split('/__tests__')[0]);
```
Comment (10xLaCroixDrinker) [inline packages/one-app-bundler/__tests__/webpack/module/webpack.client.spec.js:22]: Is this line meant to be duplicated?
Will the slash prevent this test from working on windows?

File: packages/one-app-bundler/package.json [hunk 3]
```diff
@@ -55,7 +55,7 @@
     "file-loader": "^6.0.0",
     "folder-hash": "^3.3.0",
     "glob": "^7.1.3",
-    "holocron-module-register-webpack-plugin": "^1.2.0",
+    "holocron-module-register-webpack-plugin": "^1.2.2",
```
Comment (10xLaCroixDrinker) [inline packages/one-app-bundler/package.json:58]: Why didn't the yarn lockfile update? Was this done manually instead of via yarn?

**Review by 10xLaCroixDrinker — CHANGES_REQUESTED** [review]

File: packages/one-app-bundler/__tests__/webpack/module/webpack.client.spec.js (hunk 2 above)
Comment (bishnubista) [reply packages/one-app-bundler/__tests__/webpack/module/webpack.client.spec.js:22]: > Is this line meant to be duplicated?
No, it was a mistake. Fixing it.

Comment (bishnubista) [reply packages/one-app-bundler/__tests__/webpack/module/webpack.client.spec.js]: Updating to `path.sep`.

File: packages/one-app-bundler/package.json (hunk 3 above)
Comment (bishnubista) [reply packages/one-app-bundler/package.json:58]: Updated. I ran `npm` instead of `yarn`.

**Review by 10xLaCroixDrinker — APPROVED** [review]

**Review by dogpatch626 — APPROVED** [review]

**Pull request merged**